### PR TITLE
Configure monitoring and registry during OCS deployment in external mode

### PR DESCRIPTION
**Pull request by clacroix12**
Fixes: #3465

Signed-off-by: Coady LaCroix <clacroix@redhat.com>

### CODE REVIEW

File: ocs_ci/deployment/deployment.py
Comment (sidhant-agrawal) [inline ocs_ci/deployment/deployment.py:826]: why are we skipping registry change for external mode?

Comment (clacroix12) [reply ocs_ci/deployment/deployment.py:826]: I added this `if` because of the failure in the first PR validation. Should we still be running this? If so it looks like we will need to change the behavior in this function.

Comment (sidhant-agrawal) [reply ocs_ci/deployment/deployment.py:826]: Yes, we should be running this.
We need to use appropriate value of `sc_name` here [change_registry_backend_to_ocs()](https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/registry.py#L28)
Similar to what we are doing in `setup_persistent_monitoring()`

Comment (clacroix12) [reply ocs_ci/deployment/deployment.py]: Done in latest commit

File: ocs_ci/deployment/deployment.py
Comment (shylesh) [inline ocs_ci/deployment/deployment.py:1039]: @clacroix12 , Instead of sleep can we wait for the resource? rest LGTM.

Comment (clacroix12) [reply ocs_ci/deployment/deployment.py:1039]: This logic hasn't changed from before, I just moved it into a new function to clean up a bit. Regardless, isn't that what `validate_pods_are_respinned_and_running_state` should be doing already?

Comment (shylesh) [reply ocs_ci/deployment/deployment.py:1039]: ok so if that is the case then do we need sleep here ?

Comment (shylesh) [reply ocs_ci/deployment/deployment.py:1039]: since it was already existing , I don't want to hold this PR. LGTM.

**Review by shylesh — APPROVED** [review]
+1

**Review by sidhant-agrawal — APPROVED** [review]
LGTM

**Comment by clacroix12** [comment]
Raised an issue for the failure blocking PR validation here: #3789

**Comment by clacroix12** [comment]
> Raised an issue for the failure blocking PR validation here: #3789

This is actually due to a known bug: https://bugzilla.redhat.com/show_bug.cgi?id=1921864

I am kicking off a deployment test to verify that non-external mode deployments are okay after these changes.

**Pull request merged**
